client/state/activity-log/log/reducer.js

```diff
@@ -1,18 +1,44 @@
 /** @format */
 /**
+ * External dependencies
+ */
+import { mapValues } from 'lodash';
+
+/**
  * Internal dependencies
  */
+import ActivityQueryManager from 'lib/query-manager/activity';
+import { ACTIVITY_LOG_UPDATE, DESERIALIZE, SERIALIZE } from 'state/action-types';
+import { createReducer, keyedReducer, isValidStateWithSchema } from 'state/utils';
+
 import { logItemsSchema } from './schema';
-import { ACTIVITY_LOG_UPDATE } from 'state/action-types';
-import { createReducer, keyedReducer } from 'state/utils';
 
-// FIXME: No-op reducers
 export const logError = keyedReducer( 'siteId', state => state );
 
-export const logItems = keyedReducer(
-	'siteId',
-	createReducer( undefined, {
-		[ ACTIVITY_LOG_UPDATE ]: ( state, { data } ) => data,
-	} )
+export const logItems = createReducer(
+	{},
+	{
+		[ ACTIVITY_LOG_UPDATE ]: ( state, { data, found, query, siteId } ) => {
+			return state[ siteId ]
+				? {
+						...state,
+						[ siteId ]: state[ siteId ].receive( data, { found, query } ),
+					}
+				: {
+						...state,
+						[ siteId ]: new ActivityQueryManager().receive( data, { found, query } ),
+					};
+		},
+		[ DESERIALIZE ]: state => {
+			if ( ! isValidStateWithSchema( state, logItemsSchema ) ) {
+				return {};
+			}
+			return mapValues( state, ( { data, options } ) => {
+				return new ActivityQueryManager( data, options );
+			} );
+		},
+		[ SERIALIZE ]: state => {
+			return mapValues( state, ( { data, options } ) => ( { data, options } ) );
```

Comment:
Had some discussion with @sirreal around whether there could be `serialize`/`deserialize` methods on `ActivityQueryManager`. Questionable whether there would be any benefit to changing.

Reply:
Just leaving a trail. Related conversation: p1506425123000229-slack-luna

[See `keyedReducer` with custom persistence implementation here](https://github.com/Automattic/wp-calypso/pull/18944/files#diff-d2abea61608b55b1d3a1fa81ea264e55)